cmd/minishift/cmd/start_preflight.go

```diff
@@ -414,3 +422,31 @@ func isMounted(driver drivers.Driver, mountpoint string) (bool, error) {
 
 	return true, nil
 }
+
+// checkIsoUrl checks the Iso url and returns true if the iso file exists
+func checkIsoUrl() bool {
+	isoUrl := viper.GetString(configCmd.ISOUrl.Name)
+
+	if isoUrl == "b2d" || isoUrl == "centos" || isoUrl == "minikube" {
+		return true
+	}
+
+	if runtime.GOOS == "windows" {
```

Comment:
do we really need a OS check. Can we not check as single regexp?

Reply:
But a path like `file:///home/anath/shared` will be valid in a linux host but not in a windows host, so if we don't check for the OS and put it in the same regexp, will it work as intended, because it will match any one of the pattern.

Reply:
The check for windows path is file://x:/ and for other platforms is file:///home/users/

I haven't check the code yet, but I guess this is whybhe checks for the OS